### PR TITLE
set_title when calling #view, like in rodauth

### LINKED ISSUE #112: rodauth-rails omits setting title when calling the view
`rodauth-rails`, when overriding [view](https://github.com/janko/rodauth-rails/blob/master/lib/rodauth/rails/feature/render.rb#L12), skips the [set_title](https://github.com/jeremyevans/rodauth/blob/f6c4fb1ba03901d3d646cdd7a4bf6d7bce96528d/lib/rodauth/features/base.rb#L388) logic, which means that `@page_title` won't be set. I believe this is a bug, at least considering someone using the `title_instance_variable` option.

### PR DESCRIPTION
**Pull request by HoneyryderChuck**
Closes #112

### CODE REVIEW

File: lib/rodauth/rails/feature/render.rb [hunk 1]
```diff
@@ -50,6 +51,12 @@ def _view(meth, *)
           html = html.gsub(/<form(.+)>/, '<form\1 data-turbo="false">') if meth == :view
           html
         end
+
+        def set_title(title)
+          if title_instance_variable
+            _rails_controller_instance.instance_variable_set(title_instance_variable, title)
```
Comment (janko) [inline lib/rodauth/rails/feature/render.rb:57]: Here `_rails_controller_instance` is a non-cached instance of the controller, every invocation of this method returns a new instance. We want to use the cached `rails_controller_instance` method instead (generated by `auth_cached_method :rails_controller_instance`).

**Review by janko — COMMENTED** [review]
In addition to the comment below, we should add some tests as well.

File: lib/rodauth/rails/feature/render.rb (hunk 1 above)
Comment (HoneyryderChuck) [reply lib/rodauth/rails/feature/render.rb]: done 👍

**Comment by HoneyryderChuck** [comment]
@janko added a test. I didn't find a way to "dynamically" set rodauth configuration before the test, so I went with creating a separate test app instead. Let me know if this works for you.

**Comment by janko** [comment]
@HoneyryderChuck I was thinking we could just set it on `RodauthMain`, and then tests exercising `RodauthMain` will already test rendering without `title_instance_variable` being set.

**Comment by HoneyryderChuck** [comment]
But if we set in on main, then the default will be overwritten all the time, so you won't test the "no variable set" path. Is that acceptable?

**Comment by janko** [comment]
It will be set for main, but not admin, as far as I could tell. So, we should be able to test the unset instance variable by visiting an admin route.

**Comment by HoneyryderChuck** [comment]
@janko done, let me know if this works out.

**Comment by janko** [comment]
Looks great, thanks for updating 👍🏻

**Pull request merged**